src/setPath.ts

```diff
@@ -1,11 +1,31 @@
-import type { Narrow } from "./_narrow";
-import type { Path, SupportsValueAtPath, ValueAtPath } from "./_paths";
 import { purry } from "./purry";
 
+type Path<T, Prefix extends ReadonlyArray<unknown> = readonly []> =
```

Comment:
**question:** maybe we should share these types with `pathOr`?

Reply:
There is a back-and-forth between overgeneralizing and repeating ourselves. These types were previously generic, widened to fit "the generic case," but only used in one place, and `pathOr` (and I assume other functions with path semantics) built their versions. This results in bad typing for the overgeneralized type and code deterioration for the latter almost identical type whenever a bug is fixed.

My approach to these kinds of things is to keep the types close to the function that introduced them until someone actively wants to re-use them for another function; when that happens, they can pull them into  _types.ts or somewhere else that makes sense and generalize them so they fit the 2 cases (but only those 2).

I'm not a huge fan of the path-related functions so I don't want to overwork trying to generalize the types here. I got here because I was cleaning up and wanted to remove any unused abstractions.